docusaurus/docs/React/custom-code-examples/channel-search.mdx

```diff
@@ -0,0 +1,313 @@
+
+## How-to Guide for Customizing ChannelSearch
+
+In this example we will customize the [`ChannelSearch`](../core-components/channel_list/#channelsearch) component. Though this component can be used standalone, we will do our customization when it's established in the `ChannelList`. Here is a guide that outlines how to
+override the default `DropdownContainer` and `SearchResultItem` components, as well as use several other useful props.
+
+:::note
+A complete override of this component is possible by utilizing the [`ChannelSearch`](../core-components/channel_list/#channelsearch) prop on `ChannelList`.
+:::
+
+### Step 1 - Setting Up For Success
+
+To make the search component visible in the `ChannelList`, you need to use the `showChannelSearch` prop. At this point, the search component is ready to go with all of the default UI and functionality.
+
+Let's allow the query to search for `channels` in addition to the default query of just users by setting the `searchForChannels` prop. You'll notice that all props for `ChannelSearch`
+are drilled to the component via the `additionalChannelSearchProps` prop on `ChannelList`. With just this one additional prop, the search functionality is fully functional and expanded from just the default search of users.
+
+```jsx
+<ChannelList
+    showChannelSearch
+    additionalChannelSearchProps={{searchForChannels: true }}
+/>
+```
+
+### Step 2 - Customizing the DropdownContainer
+
+Since we've enabled the search to also search for channels, let's customize the dropdown results container to fit a design that separates the result by type.
+
+```jsx
+  const CustomDropdown = (props: DropdownContainerProps) => {
+    const { results, focusedUser, selectResult, SearchResultItem } = props;
+
+    let items: ChannelOrUserResponse[] = results.filter(x => x.cid);
+    let users: ChannelOrUserResponse[] = results.filter(x => !x.cid);
+
+    return (
+      <div>
+        <p>Channels</p>
+        {!items.length && <p>No Channels...</p>}
+        {items.map((result: ChannelOrUserResponse, index: number) => (
+          <SearchResultItem
+            focusedUser={focusedUser}
+            index={index}
+            key={index}
+            result={result}
+            selectResult={selectResult}
+          />
+        ))}
+        <p>Users</p>
+        {!users.length && <p>No Users...</p>}
+        {users.map((result: ChannelOrUserResponse, index: number) => (
+          <SearchResultItem
+            focusedUser={focusedUser}
+            index={index}
+            key={index}
+            result={result}
+            selectResult={selectResult}
+          />
+        ))}
+      </div>
+    );
+  };
+```
+
+```jsx
+const DropDown = (props: DropdownContainerProps) => <CustomDropdown {...props} />;
+
+<ChannelList
+    showChannelSearch
+    additionalChannelSearchProps={
+        {
+          searchForChannels: true,
+          DropdownContainer: DropDown
+        }
```

Comment:
we should pull these objects out into consts to show best practices, having a constant reference will save renders